apps/frontend/src/pages/CreateCharity.tsx

```diff
@@ -0,0 +1,651 @@
+        console.error(
+          "newCharityObject.organizations is not an array:",
+          newCharityObject.organizations,
+        );
+        return false;
+      }
+
+      // Upload partnerImages
+      const partnerImagesArray = partnerImages
+        .filter((obj) => obj.image !== undefined)
+        .map((obj, index) => ({ image: obj.image as File, index }));
+
+      const partnerImagesObjectArray =
+        await asyncUploadImages(partnerImagesArray);
+      if (partnerImagesObjectArray && partnerImagesObjectArray.length > 0) {
+        const newOrganizations = [...newCharityObject.organizations];
+        for (let i = 0; i < newCharityObject.organizations.length; i++) {
+          newOrganizations[partnerImagesObjectArray[i].index].logoUrl =
+            partnerImagesObjectArray[i].url;
+        }
+        setNewCharityObject((prevState) => ({
+          ...prevState,
+          organizations: newOrganizations,
+        }));
+        return {
+          ...newCharityObject,
+          organizations: newOrganizations,
+          imageUrl: logoUrlResult as string,
+        };
+      } else {
+        console.error("No images uploaded for partnerImages.");
+        return false;
+      }
+    } catch (error) {
+      console.error("Error uploading partner images:", error);
+      return false;
+    }
+  };
+
+  // Uploads the payload
+  const updateNewCharityPayload = (key: keyof CharityObject, value: string) => {
+    setNewCharityObject((prevState) => ({
+      ...prevState,
+      [key]: value,
+    }));
+  };
+
+  // Updates the number of organizations or updates them if they already exist
+  const updateOrganizationPayload = (
+    index: number,
+    key: keyof OrganizationObject,
+    value: string | number | boolean,
+  ) => {
+    setNewCharityObject((prevState) => {
+      const updatedOrganizations = [...prevState.organizations];
+      try {
+        updatedOrganizations[index] = {
+          ...updatedOrganizations[index],
+          [key]: value,
+        };
+      } catch (error) {
+        const newOrganization: OrganizationObject = {
+          name: "",
+          logoUrl: "",
+          donated: 0,
+          receiving: false,
+        };
+        updatedOrganizations.push(newOrganization);
+      }
+
+      return {
+        ...prevState,
+        organizations: updatedOrganizations,
+      };
+    });
+  };
+
+  const updateCharityName = (event: ChangeEvent<HTMLInputElement>) => {
+    updateNewCharityPayload("name", event.target.value);
+  };
+
+  const updateDescription = (event: ChangeEvent<HTMLInputElement>) => {
+    updateNewCharityPayload("description", event.target.value);
+  };
+
+  const updateStartDate = (event: ChangeEvent<HTMLInputElement>) => {
+    const dateValue = event.target.value;
+    setStartDate(dateValue);
+    if (dateValue) {
+      const dateStart = new Date(dateValue);
+      const dateEnd = new Date(dateStart);
+      dateEnd.setDate(dateStart.getDate() + 31);
```

Comment:
This won't work on months with less than 31 days, or if we wanted to just create 12 charities off the bat for a full year. We should probably be setting the end date manually as well (this page should never be exposed to a user anyways, its more for us to add charities)

Reply:
Re-added the end date and added form validation to address this so you guys can create whatever you'd like